airflow/providers/amazon/aws/operators/aws_lambda.py

```diff
@@ -0,0 +1,121 @@
+# under the License.
+
+import json
+from typing import TYPE_CHECKING, Optional, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.lambda_function import LambdaHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class AwsLambdaInvokeFunctionOperator(BaseOperator):
+    """
+    Invokes an AWS Lambda function.
+    You can invoke a function synchronously (and wait for the response),
+    or asynchronously.
+    To invoke a function asynchronously,
+    set InvocationType to `Event`.
+
+    :param function_name: The name of the AWS Lambda function, version, or alias.
+    :param payload: The JSON string that you want to provide to your Lambda function as input.
+    :param log_type: Set to Tail to include the execution log in the response. Otherwise, set to "None".
+    :param qualifier: Specify a version or alias to invoke a published version of the function.
+    :param aws_conn_id: The AWS connection ID to use
+    """
+
+    template_fields: Sequence[str] = ('function_name', 'payload', 'qualifier', 'invocation_type')
+    ui_color = '#ff7300'
+
+    def __init__(
+        self,
+        *,
+        function_name: str,
+        log_type: Optional[str] = None,
+        qualifier: Optional[str] = None,
+        invocation_type: Optional[str] = None,
+        client_context: Optional[str] = None,
+        payload: Optional[str] = None,
+        aws_conn_id='aws_default',
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.FunctionName = function_name
+        self.Payload = payload
+        self.LogType = log_type
+        self.Qualifier = qualifier
+        self.InvocationType = invocation_type
+        self.ClientContext = client_context
+        self.aws_conn_id = aws_conn_id
+
+    def execute(self, context: 'Context'):
+        """
+        Invokes the target AWS Lambda function from Airflow.
+
+        :return: The response payload from the function, or an error object.
+        """
+        hook = LambdaHook(aws_conn_id=self.aws_conn_id)
+        success_status_codes = [200, 202, 204]
+        self.log.info("Invoking AWS Lambda function: %s with payload: %s", self.FunctionName, self.Payload)
+        try:
+            response = hook.invoke_lambda(
+                function_name=self.FunctionName,
+                **{
+                    key: self.__dict__[key]
+                    for key in self.__dict__
+                    if (
+                        key
+                        in [
+                            "Payload",
+                            "LogType",
+                            "Qualifier",
+                            "InvocationType",
+                            "ClientContext",
+                        ]
+                        and self.__dict__[key] is not None
+                    )
+                },
+            )
+            self.log.info("Lambda response metadata: %s", json.dumps(response.get("ResponseMetadata")))
```

Comment:
I don't think you need to serialize this as JSON; a Python repr string would be fine here:
```python
self.log.info("Lambda response metadata: %r", response.get("ResponseMetadata"))
```
Also, there's a lot of repetition in the error handling here.

Reply:
@malthe, thank you for the review. I've made the changes for error handling. 

However, I am unable to think of a way to reduce the error handling bit. The errors are handled based on the response from the AWS API. If an exception is not thrown by the underlying boto3 SDK, the handling of the same becomes conditional leading to the multiple `if` blocks.

Would you have any recommendations?

Reply:
@schirag1993 I think @malthe may have been referring to the same thing as my comment, which is already resolved? https://github.com/apache/airflow/pull/21686#discussion_r810686375

Reply:
Yes, looks like it @edwardwang888. That should mark all comments addressed then.